### ISSUE TITLE
GIF commands: internal error: rootURL must be set

### ISSUE THREAD

**Issue by phpfs** [moussetc/mattermost-plugin-giphy#52]
**Describe the bug**
After startup, we get the following error:

```
[ERROR] call to OnConfigurationChange failed, error: GIF commands: internal error: rootURL must be set, ","caller":"io/io.go:428","plugin_id":"com.github.moussetc.mattermost.plugin.giphy","source":"plugin_stderr"
```

Besides, the plugin seems to work just fine.

**To Reproduce**
1. Install plugin.
2. Start Mattermost server.

**Expected behavior**
Start up without an error.

**Screenshots**

**Mattermost information**
- server version: 7.5.1
- client version 5.2.2
- plugin version 2.1.0

**Comment by phpfs** [comment]
It appears that this error is raised here...

https://github.com/moussetc/mattermost-plugin-giphy/blob/75652a72e29dca23d0af03dce5e8db4f1efc10f2/server/internal/provider/giphy.go#L51

...with the `rootUrl` field set here:

https://github.com/moussetc/mattermost-plugin-giphy/blob/75652a72e29dca23d0af03dce5e8db4f1efc10f2/server/configuration.go#L56

The `rootUrl` is populated as part of `OnActivate`:

https://github.com/moussetc/mattermost-plugin-giphy/blob/75652a72e29dca23d0af03dce5e8db4f1efc10f2/server/plugin.go#L49

Any idea why this field could stay completely empty?

**Comment by moussetc** [comment]
@phpfs is the Mattermost Site URL properly set as explained in [this documentation](https://docs.mattermost.com/install/config-mattermost-server.html)?

**Comment by phpfs** [comment]
> @phpfs is the Mattermost Site URL properly set as explained in [this documentation](https://docs.mattermost.com/install/config-mattermost-server.html)?

Yes, it is the public facing domain name with correct protocol - and validated using the `Test Live URL` button.

**Comment by phpfs** [comment]
FYI: I just noticed that since we run 2.1.0 (marketplace version), the changes of #50 are not yet in. Hence, the `siteUrl` is not yet used. @moussetc

**Comment by moussetc** [comment]
> Any idea why this field could stay completely empty?

The only way I see would be for `onConfigurationChange()` to be called before `onActivate()`. Which, uh, I guess can happen in a plugin lifecycle?

Btw, thank you for the report and trying to find the cause in the code, much appreciated. I don't plan on fixing this soon as it is a non-critical bug, but now that the root cause is known, I will fix it in the next version (whenever that would be, I don't have a release schedule :sweat_smile: ).

**Comment by moussetc** [comment]
Fixes in 3.0.0
